File: frontend/src/api/groups.ts
/** 分组相关 API */
import client, { API_PREFIX } from './client';
import type { Group } from '@/types';

export const getGroupList = (search?: string) => {
  const params = new URLSearchParams();
  if (search) params.append('search', search);
  const qs = params.toString();
  return client.get<{ groups: Group[] }>(`${API_PREFIX}/groups${qs ? `?${qs}` : ''}`);
};

export const getGroup = (id: number) =>
  client.get<Group>(`${API_PREFIX}/groups/${id}`);

export const createGroup = (data: { name: string; notes?: string }) =>
  client.post<{ id: number; message: string }>(`${API_PREFIX}/groups`, data);

export const updateGroup = (id: number, data: { name: string; main_account_id?: number; notes?: string }) =>
  client.put<{ message: string }>(`${API_PREFIX}/groups/${id}`, data);

export const deleteGroup = (id: number) =>
  client.delete<{ message: string }>(`${API_PREFIX}/groups/${id}`);

export const addAccountToGroup = (groupId: number, accountId: number) =>
  client.post<{ message: string }>(`${API_PREFIX}/groups/${groupId}/accounts/${accountId}`);

export const removeAccountFromGroup = (accountId: number) =>
  client.delete<{ message: string }>(`${API_PREFIX}/groups/accounts/${accountId}`);

export const setMainAccount = (groupId: number, accountId: number) =>
  client.put<{ message: string }>(`${API_PREFIX}/groups/${groupId}/main-account/${accountId}`);

// ── 号池管理 ──

export const getPoolAccounts = (groupId: number) =>
  client.get<{ accounts: import('@/types').Account[] }>(`${API_PREFIX}/groups/${groupId}/pool`);

export const addToPool = (groupId: number, accountIds: number[]) =>
  client.post<{ message: string }>(`${API_PREFIX}/groups/${groupId}/pool`, { account_ids: accountIds });

export const removeFromPool = (groupId: number, accountIds: number[]) =>
  client.delete<{ message: string }>(`${API_PREFIX}/groups/${groupId}/pool`, { data: { account_ids: accountIds } });

File: backend/routers/groups.py
"""分组路由 - 分组 CRUD、成员管理"""
from fastapi import APIRouter, Depends, HTTPException, status

from deps import get_group_service, verify_token
from models.schemas import GroupCreate, GroupUpdate
from services.group import GroupService

router = APIRouter(
    prefix="/groups", tags=["分组"], dependencies=[Depends(verify_token)]
)


@router.get("")
def list_groups(
    search: str = "",
    svc: GroupService = Depends(get_group_service),
):
    """获取所有分组 (支持按子号邮箱搜索)"""
    return {"groups": svc.get_all(search=search)}


@router.get("/{group_id}")
def get_group(group_id: int, svc: GroupService = Depends(get_group_service)):
    """获取分组详情（含成员）"""
    group = svc.get_with_accounts(group_id)
    if not group:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="分组不存在")
    return group


@router.post("", status_code=status.HTTP_201_CREATED)
def create_group(
    data: GroupCreate,
    svc: GroupService = Depends(get_group_service),
):
    """创建分组"""
    group_id = svc.create(name=data.name, notes=data.notes)
    return {"id": group_id, "message": "分组创建成功"}


@router.put("/{group_id}")
def update_group(
    group_id: int,
    data: GroupUpdate,
    svc: GroupService = Depends(get_group_service),
):
    """更新分组"""
    svc.update(
        group_id=group_id,
        name=data.name,
        main_account_id=data.main_account_id,
        notes=data.notes,
    )
    return {"message": "分组更新成功"}


@router.delete("/{group_id}")
def delete_group(
    group_id: int, svc: GroupService = Depends(get_group_service)
):
    """删除分组"""
    svc.delete(group_id)
    return {"message": "分组删除成功"}


@router.post("/{group_id}/accounts/{account_id}")
def add_account(
    group_id: int,
    account_id: int,
    svc: GroupService = Depends(get_group_service),
):
    """将账号添加到分组"""
    try:
        svc.add_account(group_id, account_id)
        return {"message": "账号已添加到分组"}
    except ValueError as e:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))


@router.delete("/accounts/{account_id}")
def remove_account(
    account_id: int,
    svc: GroupService = Depends(get_group_service),
):
    """将账号从分组移除"""
    svc.remove_account(account_id)
    return {"message": "账号已从分组移除"}


@router.put("/{group_id}/main-account/{account_id}")
def set_main_account(
    group_id: int,
    account_id: int,
    svc: GroupService = Depends(get_group_service),
):
    """设置分组主号"""
    try:
        svc.set_main_account(group_id, account_id)
        return {"message": "主号设置成功"}
    except ValueError as e:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))
